### PR TITLE
Elasticsearch: Add tracking for plugin adoption stats

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What is this feature?**

This PR adds tracking to all queries that are ran through the ES datasource. Also it adds tracking for ES plugin adoption stats.

**Why do we need this feature?**

Learn about how users use ES.

### CODE REVIEW

**Review by svennergr — COMMENTED** [review]
Nice, left a few comments.

File: public/app/plugins/datasource/elasticsearch/tracking.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+  if (nonMetricQueryTypes.includes(query.metrics[0].type)) {
+    return query.metrics[0].type;
+  }
+  return 'metric';
+};
+
+const getLineLimit = (query: ElasticsearchQuery): number | undefined => {
+  // We only want to track line limit for log queries
+  const logMetric = query.metrics?.[0]?.type === 'logs' ? query.metrics[0] : undefined;
+  const lineLimitString = logMetric ? logMetric.settings?.limit : undefined;
```
[the remark this answers is not in the document]
Comment (ivanahuckova) [reply public/app/plugins/datasource/elasticsearch/tracking.ts]: Changed in https://github.com/grafana/grafana/pull/59954/commits/f7b0f79d4b1998bacc3edb811c8c9e91ae3d46b9. With `getQueryType`, typescript had issue (our query types in tracking are simplified types), but I've refactored it to escape early and make it more readable.

**Review by svennergr — APPROVED** [review]
LGTM 🚀

File: public/app/plugins/datasource/elasticsearch/tracking.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
+import { DashboardLoadedEvent, DataQueryResponse } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+import { variableRegex } from 'app/features/variables/utils';
+
+import pluginJson from './plugin.json';
+import { ElasticsearchQuery } from './types';
+
+type LokiOnDashboardLoadedTrackingEvent = {
```
Comment (svennergr) [inline public/app/plugins/datasource/elasticsearch/tracking.ts:8]:
```suggestion
type ElasticSearchOnDashboardLoadedTrackingEvent = {
```

**Review by svennergr — CHANGES_REQUESTED** [review]

File: public/app/plugins/datasource/elasticsearch/tracking.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    const queriesWithTemplateVariables = elasticsearchQueries.filter(isQueryWithTemplateVariables);
+    const queriesWithLuceneQuery = elasticsearchQueries.filter((query) => !!query.query);
+    const logsQueries = elasticsearchQueries.filter((query) => getQueryType(query) === 'logs');
+    const metricQueries = elasticsearchQueries.filter((query) => getQueryType(query) === 'metric');
+    const rawDataQueries = elasticsearchQueries.filter((query) => getQueryType(query) === 'raw_data');
+    const rawDocumentQueries = elasticsearchQueries.filter((query) => getQueryType(query) === 'raw_document');
+    const queriesWithChangedLineLimit = elasticsearchQueries.filter(isQueryWithChangedLineLimit);
+
+    const event: LokiOnDashboardLoadedTrackingEvent = {
```
Comment (svennergr) [inline public/app/plugins/datasource/elasticsearch/tracking.ts:56]:
```suggestion
    const event: ElasticSearchOnDashboardLoadedTrackingEvent = {
```

**Review by svennergr — APPROVED** [review]

**Comment by ivanahuckova** [comment]
🙈 fixed

**Pull request merged**